packages/query-core/src/__tests__/queryObserver.test.tsx

```diff
@@ -925,42 +925,6 @@ describe('queryObserver', () => {
     expect(observer.getCurrentResult().data).toBe(selectedData2)
   })
 
-  test('should not use an undefined value returned by select as placeholderData', () => {
-    const key = queryKey()
-
-    const data = { value: 'data' }
-    const selectedData = { value: 'data' }
-    const placeholderData1 = { value: 'data' }
-    const placeholderData2 = { value: 'data' }
-
-    const observer = new QueryObserver(queryClient, {
-      queryKey: key,
-      queryFn: () => data,
-      select: () => data,
-    })
-
-    observer.setOptions({
-      queryKey: key,
-      queryFn: () => data,
-      select: () => {
-        return selectedData
-      },
-      placeholderData: placeholderData1,
-    })
-
-    expect(observer.getCurrentResult().isPlaceholderData).toBe(true)
-
-    observer.setOptions({
-      queryKey: key,
-      queryFn: () => data,
-      // @ts-expect-error
-      select: () => undefined,
```

Comment:
note: this test was quite weird to begin with because it’s not allowed to return undefined from select 🤷